Sources/Nimble/Matchers/SatisfyAllOf.swift

```diff
@@ -12,7 +12,7 @@ public func satisfyAllOf<T, U>(_ matchers: U...) -> Predicate<T>
         return satisfyAllOf(matchers.map { $0.predicate })
 }
 
-internal func satisfyAllOf<T>(_ predicates: [Predicate<T>]) -> Predicate<T> {
+public func satisfyAllOf<T>(_ predicates: [Predicate<T>]) -> Predicate<T> {
```

Comment:
```suggestion
/// A Nimble matcher that succeeds when the actual value matches with all of the matchers
/// provided in the variable list of matchers.
public func satisfyAllOf<T>(_ predicates: [Predicate<T>]) -> Predicate<T> {
```

Let's add the doc comment.